flink-end-to-end-tests/flink-end-to-end-tests-sql/src/test/java/org/apache/flink/table/sql/codegen/SQLCodegenE2EITCase.java

```diff
@@ -0,0 +1,198 @@
+
+            List<String> sqlLines = initializeSqlLines(varsMap);
+
+            // Execute SQL statements in "codegen_e2e.sql" file
+            executeSqlStatements(clusterController, sqlLines);
+
+            // Wait until all the results flushed to the json file.
+            LOG.info("Verify the json result.");
+            checkJsonResultFile();
+            LOG.info("The codegen SQL client test run successfully.");
+        }
+    }
+
+    private void executeSqlStatements(ClusterController clusterController, List<String> sqlLines)
+            throws IOException {
+        LOG.info("Executing end-to-end SQL statements {}.", sqlLines);
+        clusterController.submitSQLJob(
+                new SQLJobSubmission.SQLJobSubmissionBuilder(sqlLines)
+                        .addJar(sqlToolBoxJar)
+                        .build(),
+                Duration.ofMinutes(2L));
+    }
+
+    private List<String> initializeSqlLines(Map<String, String> vars) throws IOException {
+        URL url = SQLCodegenE2EITCase.class.getClassLoader().getResource(CODEGEN_E2E_SQL);
+        if (url == null) {
+            throw new FileNotFoundException(CODEGEN_E2E_SQL);
+        }
+
+        List<String> lines = Files.readAllLines(new File(url.getFile()).toPath());
+        List<String> result = new ArrayList<>();
+        for (String line : lines) {
+            for (Map.Entry<String, String> var : vars.entrySet()) {
+                line = line.replace(var.getKey(), var.getValue());
+            }
+            result.add(line);
+        }
+
+        return result;
+    }
+
+    private void checkJsonResultFile() throws Exception {
+        boolean success = false;
+        final Deadline deadline = Deadline.fromNow(Duration.ofSeconds(20));
+        while (deadline.hasTimeLeft()) {
+            if (Files.exists(result)) {
+                List<String> lines = readJsonResultFiles(result);
+                if (lines.size() == 2) {
+                    success = true;
+                    assertThat(
+                            lines.toArray(new String[0]),
+                            arrayContainingInAnyOrder(
+                                    "{\"before\":null,\"after\":{\"user_name\":\"Alice\",\"order_cnt\":1},\"op\":\"c\"}",
+                                    "{\"before\":null,\"after\":{\"user_name\":\"Bob\",\"order_cnt\":2},\"op\":\"c\"}"));
+                    break;
+                } else {
+                    LOG.info(
+                            "The target Json {} does not contain enough records, current {} records, left time: {}s",
+                            result,
+                            lines.size(),
+                            deadline.timeLeft().getSeconds());
+                }
+            } else {
+                LOG.info("The target Json {} does not exist now", result);
+            }
+            Thread.sleep(500);
+        }
+        Assert.assertTrue("Did not get expected results before timeout.", success);
```

Comment:
Please print the content of the result file if there is any. It would help to debug if the test is not passed on Azure.